src/main/java/uk/ac/sanger/sccp/stan/service/label/LabwareLabelData.java

```diff
@@ -56,19 +64,23 @@ public int hashCode() {
 
     @Override
     public String toString() {
-        return MoreObjects.toStringHelper(this)
+        return BasicUtils.describe(this)
                 .add("barcode", barcode)
+                .add("externalBarcode", externalBarcode)
                 .add("medium", medium)
                 .add("date", date)
                 .add("contents", contents)
+                .omitNullValues()
+                .reprStringValues()
                 .toString();
     }
 
     public Map<String, String> getFields() {
-        HashMap<String, String> fields = new HashMap<>(3 + 4 * contents.size());
+        HashMap<String, String> fields = new HashMap<>(4 + 4 * contents.size());
         fields.put("barcode", getBarcode());
```

Comment:
For future references could you please create global variables to denote what these values refer too